src/Lean/Server/FileWorker.lean

```diff
@@ -90,7 +90,7 @@ section Elab
       : AsyncElabM (Option Snapshot) := do
     cancelTk.check
     let s ← get
-    let lastSnap := s.snaps.back
+    let .some lastSnap := s.snaps.back? | panic! "empty snapshots"
```

Comment:
`AsyncElabM` is an exception monad, why not `throw`?

Reply:
Throwing a string is a bit cumbersome here, you'd need `throw (.ioError (.userError "empty snapshots"))`.  Ultimately, I think the effect is the same: you see an error in the editor.

Reply:
You can write `throwThe ElabTaskError "empty snapshots"`, I think. The thing that I find weird about the `panic!` is that in case it happens, we go on to execute the `Inhabited` instance, right? If I'm reading `#synth` right that ends up being `pure none` which is a lot of implicit magic. I would agree with Mario that `panic!` is best avoided, although would say more weakly that avoiding is not always worth it when there isn't already an exception effect available (but there is here).

Reply:
Let's not change panic usage policy in this PR. I would say the current policy is to use panics for internal logic errors, which is true for this instance.